Makefile

```diff
@@ -191,6 +189,9 @@ deploy-mutation: patch-image
 
 # Deploy controller in the configured Kubernetes cluster in ~/.kube/config
 deploy: patch-image manifests
+ifeq ($(ENABLE_EXTERNAL_DATA),true)
+	@grep -q -v 'enable-external-data' ./config/overlays/dev/manager_image_patch.yaml && sed -i '/- --operation=webhook/a \ \ \ \ \ \ \ \ - --enable-external-data=true' ./config/overlays/dev/manager_image_patch.yaml
```

Comment:
Do we want to create a separate target like `deploy-mutation` and move this there without any conditionals?

Reply:
i think conditions are better otherwise we end up having 2 different combinations: validation (w/ and w/o external data) and mutation (w/ and w/o external data).